fineract-core/src/main/java/org/apache/fineract/accounting/producttoaccountmapping/domain/PortfolioProductType.java

```diff
@@ -47,33 +44,41 @@ public String getCode() {
         return this.code;
     }
 
-    private static final Map<Integer, PortfolioProductType> intToEnumMap = new HashMap<>();
-
-    static {
-        for (final PortfolioProductType type : PortfolioProductType.values()) {
-            intToEnumMap.put(type.value, type);
+    public static PortfolioProductType fromInt(final Integer v) {
+        if (v == null) {
+            return null;
         }
-    }
 
-    public static PortfolioProductType fromInt(final int i) {
-        final PortfolioProductType type = intToEnumMap.get(Integer.valueOf(i));
-        return type;
+        switch (v) {
```

Comment:
Please do not hardcode the mapping, same as for GLAccountType. I do not comment other enums but this request applies for all of them.

Reply:
See comment above.